packages/core/src/crawlers/error_snapshotter.ts

```diff
@@ -0,0 +1,148 @@
+        const defaultStorageDir = () => {
+            if (pathExistsSync(resolve('./crawlee_storage'))) {
+                return 'crawlee_storage';
+            }
+
+            return 'storage';
+        };
+
+        this.KEY_VALUE_PLATFORM_PATH = 'https://api.apify.com/v2/key-value-stores';
+        this.KEY_VALUE_STORE_LOCAL_PATH = `file://${process.env.PWD}/${defaultStorageDir()}/key_value_stores`;
+    }
+
+    /**
+     * Capture a snapshot of the error context.
+     */
+    async captureSnapshot(error: ErrnoException, context: CrawlingContext): Promise<{ screenshotFileUrl?: string; htmlFileUrl?: string }> {
+        try {
+            const page = context?.page as PuppeteerPage | PlaywrightPage | undefined;
+            const body = context?.body;
+
+            const keyValueStore = await context?.getKeyValueStore();
+            // If the key-value store is not available, or the body and page are not available, return empty filenames
+            if (!keyValueStore || (!body && !page)) {
+                return {};
+            }
+
+            const filename = this.generateFilename(error);
+
+            let screenshotFilename: string | undefined;
+            let htmlFileName: string | undefined;
+
+            if (page) {
+                const capturedFiles = await this.captureSnapShot(
+                    context as unknown as PlaywrightCrawlingContext | PuppeteerCrawlingContext,
+                    filename,
+                );
+
+                if (capturedFiles) {
+                    screenshotFilename = capturedFiles.screenshotFilename;
+                    htmlFileName = capturedFiles.htmlFileName;
+                }
+
+                // If the snapshot for browsers failed to capture the HTML, try to capture it from the page content
+                if (!htmlFileName) {
+                    const html = await page?.content() || undefined;
+                    htmlFileName = html ? await this.saveHTMLSnapshot(html, keyValueStore, filename) : undefined;
+                }
+            } else if (typeof body === 'string') { // for non-browser contexts
+                htmlFileName = await this.saveHTMLSnapshot(body, keyValueStore, filename);
+            }
+
+            if (process.env.APIFY_IS_AT_HOME) {
+                const platformPath = `${this.KEY_VALUE_PLATFORM_PATH}/${keyValueStore.id}/records`;
+                return {
+                    screenshotFileUrl: screenshotFilename ? `${platformPath}/${screenshotFilename}` : undefined,
+                    htmlFileUrl: htmlFileName ? `${platformPath}/${htmlFileName}` : undefined,
+                };
+            }
+
+            const localPath = `${this.KEY_VALUE_STORE_LOCAL_PATH}/${keyValueStore.name || 'default'}`;
+            return {
+                screenshotFileUrl: screenshotFilename ? `${localPath}/${screenshotFilename}` : undefined,
+                htmlFileUrl: htmlFileName ? `${localPath}/${htmlFileName}` : undefined,
+            };
```

Comment:
FYI we have this `KVS.getPublicUrl()` method defined in the SDK variant of the class, i would rather see an approach in this direction instead of hardcoding things on wrong level.

https://github.com/apify/apify-sdk-js/blob/master/packages/apify/src/key_value_store.ts#L12

i don't understand why would you need to care about local paths, but we could define the same method on the crawlee variant of the KVS class to resolve to that, sounds like a much cleaner solution